### DISCUSSION TITLE
Lab 5.01 task 1

### DISCUSSION THREAD

**Discussion by imEag** [rramosp/ai4eng.v1#113] [05 - GESTIÓN DE MODELOS]
Necesito ayuda con el task 1 del lab 5.01, tengo esto:

`pnl = (s[1:] - s[:-1])-s[:-1]*0.01`

y me sale incorrecto. 

Creo que no entiendo el problema, no sé como se relaciona lo que tengo con la variable p. 

gracias.

**Comment by Felipe-RA** [comment]
Saludos imEag

Para comprender este ejercicio es importante entender qué representa cada uno de los elementos que aparecen sobre él:

1. ¿Qué es un `pnl`?
2. ¿Qué representa `s`? ¿De qué está compuesta?
3. ¿Qué tipo de predicción es `p`? 
4. ¿Cuál es la utilidad de `(p-0.5)*2 `? ¿Qué hace esto y por qué lo necesitamos?

Cuentame cómo entiendes cada uno de estos elementos para que pasemos a mirar los detalles de la implementación en código de la solución.

**Comment by ManueloRZ2** [comment]
Hola un gusto saludar,

presento un problema con este punto

he creado una función como sigue para que arroje los valores del pnl:

def ResultPNL(s,p):
  vectorPnl = []
  for i in range(len(p)):
    vectorPnl.append(round(((s[i+1]-s[i])xp[i]) - (s[i]x1x.01),2))
  
  return np.array(vectorPnl)

Al probarla con el ejemplo que dan en el notebook funciona bien y al comparar con pruebas hechas a mano valor por valor también coinciden; sin embargo al tratar de enviar la respuesta me genera error de no coincidencia, ¿ existe algún formato especial para enviar la lista de pnl? ya que he visto que en puntos anteriores el redondeo puede afectar

**Reply by Felipe-RA** [reply]
Saludos MAnueloRZ2

En primer lugar, te recomiendo que utilices las ventajas que Python te ofrece para realizar operaciones vectorizadas en lugar de ciclos siempre que se pueda y tenga sentido(es más rápido, legible y más cercano a su interpretación matemática)

Ahora, con respecto a tu pregunta:

En la definición del problema tenemos que:

> fill in the pnl variable below with a <ins>**list**</ins> of 19 values corresponding on applying the same strategy as in the notes, buying or selling always ONE money unit:


Si realizas un `type(pnl)` veras que estas retornando un array de numpy


Hasta pronto!

**Reply by ManueloRZ2** [reply]
Ya hice cambio de la función que estoy usando me siguen dando los valores del ejemplo, hice el cambio a lo especificado en la actividad para que no este como un array pero aún así me arroja un o al hacer un submit de la solución.

ejemplo:
![image](https://user-images.githubusercontent.com/69865817/164934103-09938044-f0b5-4875-8b01-8526391fc103.png)

solución:
![image](https://user-images.githubusercontent.com/69865817/164934111-49d68e83-e31f-4aaa-bdc9-03f8b6071bed.png)


Código nuevo:
![image](https://user-images.githubusercontent.com/69865817/164934057-322db38a-c9ef-4642-8322-01a94d01b4fe.png)

Nuevamente uso mi s y mi p para hacer una verificación, comparo con resultados hechos a mano y coinciden, pero aun así me arroja:

![image](https://user-images.githubusercontent.com/69865817/164934202-50c691b8-16eb-4a1b-a00a-607eca1a8ef9.png)

**Reply by Felipe-RA** [reply]
¿Cómo estas obteniendo predi?

**Reply by ManueloRZ2** [reply]
le hago el siguiente tratamiento:

![image](https://user-images.githubusercontent.com/69865817/164934294-7fc9ed10-8e68-412f-a703-6235d8111c73.png)

mejor dicho, la función con los valores que arroja para mi caso el notebbok sería:

![image](https://user-images.githubusercontent.com/69865817/164934359-916439c8-3537-4050-8e10-4a22f77af5ca.png)

**Reply by Felipe-RA** [reply]
Pasame tu notebook le echo un vistazo 
(felipe.rodrigueza@udea.edu.co)

**Reply by ManueloRZ2** [reply]
Listo muchísimas gracias por el apoyo y lo oportuno, te pasé un notebook donde solo está la formula que diseñé que el otro tenia mucho borrador

**Reply by Felipe-RA** [reply]
Saludos ManueloRZ2

Disculpa la tardanza,

Tu problema esta vez es que no puedes modificar la variable p, pues esta es la que se genera aleatoriamente al principio del ejercicio.

Lo que estaba ocurriendo es lo siguiente:

1. Generación aleatoria de s y p
2. Modificas p para calcular tu pnl
3. Sometes el ejercicio (esto envía tu s, tu p y tu pnl al servidor)
4. El grader hace `p = (p-0.5)*2` con tu p ya modificada.

Este problema lo puedes solucionar poniendole otro nombre más descriptivo a la p, algo como `p_mod`, `conv_p` , etc. puede funcionarte

**Reply by ManueloRZ2** [reply]
Te agradezco muchísimo, si ya hice un cambio de variable y quedó completamente solucionado, de verdad muchas gracias y principalmente agradezco por lo oportuno de tus respuestas
